### ISSUE TITLE
Request: default workspace layout + applications

### ISSUE THREAD

**Issue by vbreder** [i3/i3#5830]
## I'm submitting a…
<pre>
[ ] Bug
[x] Feature Request
[ ] Documentation Request
[ ] Other (Please describe in detail)
</pre>

## Current Behavior
When switching to a new workspace, there is no containers or applications.

## Desired Behavior
Having a configuration parameter to set the default container tree and applications to start up on new a newly opened workspace.

## Impact
<pre>
[x] This feature requires new configuration and/or commands
</pre>

## Environment

Output of `i3 --moreversion 2>&-`:
<pre>
i3 version: 
Binary i3 version:  4.23 (2023-10-29) © 2009 Michael Stapelberg and contributors
Running i3 version: 4.23 (2023-10-29) (pid 774)
Loaded i3 config:
  /home/vb/.config/i3/config (main) (last modified: Sun 17 Dec 2023 04:53:58 PM -03, 160538 seconds ago)

The i3 binary you just called: /usr/bin/i3
The i3 binary you are running: i3
</pre>

<pre>
- Linux Distribution & Version: Arch Linux 6.6.7-arch-1-1
- Are you using a compositor (e.g., xcompmgr or compton): picom vgit-b700a
</pre>

**Comment by i3bot** [comment]
Please note that new features which require additional configuration will usually not be considered. We are happy with the feature set of i3 and want to focus in fixing bugs instead. We do accept feature requests, however, and will evaluate whether the added benefit (clearly) outweighs the complexity it adds to i3.

Keep in mind that i3 provides a powerful way to interact with it through its IPC interface: https://i3wm.org/docs/ipc.html.

**Comment by vbreder** [comment]
Note: I can implement this myself and provide a PR for your review with some minimal directions.

**Comment by okraits** [comment]
Did you take a look at:

* https://i3wm.org/docs/userguide.html#_layout_mode_for_new_containers
* https://i3wm.org/docs/layout-saving.html
* https://i3wm.org/docs/userguide.html#_automatically_starting_applications_on_i3_startup
* https://i3wm.org/docs/userguide.html#assign_workspace

i3 provides everything you need - you just have to configure it.

**Comment by vbreder** [comment]
Thanks for the quick and thoughtful reply! Yes I did look at those and I've resorted to set up, e.g., 4 workspaces at startup, with the container layout i'd like and the terminals in place.

My ask would be to, instead of the trigger for "exec" being i3 itself starting up, the trigger being having *switched* to a new workspace ("new" as in unpopulated by containers and programs).

there's "exec" (i3 startup) and "exec_always" (i3 startup or restart). maybe it would be a "exec_workspace" directive?
